types/deferred_bank_operations.go

```diff
@@ -0,0 +1,83 @@
+package types
+
+import (
+	"sort"
+	"sync"
+)
+
+type DeferredBankOperationMapping struct {
+	deferredOperations map[string]Coins
+	mappingLock		  *sync.Mutex
+}
+
+func NewDeferredBankOperationMap() *DeferredBankOperationMapping {
+	return &DeferredBankOperationMapping{
+		deferredOperations: make(map[string]Coins),
+		mappingLock: &sync.Mutex{},
+	}
+}
+
+func (m *DeferredBankOperationMapping) get(moduleAccount string) (Coins, bool) {
```

Comment:
these functions don't need locking?

Reply:
I made these private functions as they should only be called internally. let me add comments

Reply:
Actually, i just removed this. It's not really needed and adds more confusion